framework/src/base/FlagElementsThread.C

```diff
@@ -57,7 +57,10 @@ void
 FlagElementsThread::onElement(const Elem *elem)
 {
   dof_id_type dof_number = elem->dof_number(_system_number, _field_var_number, 0);
-  Marker::MarkerValue marker_value = (Marker::MarkerValue)_serialized_solution[dof_number];
+
+  // This is basically lround() but doesn't require C99 or C++11
+  Marker::MarkerValue marker_value =
+    (Marker::MarkerValue)(std::ceil(_serialized_solution[dof_number]-0.5));
```

Comment:
Still against this.  But others can outvote me.

BTW: We can at least require C99... right?